client/src/redux/fetch-user-saga.js

```diff
@@ -9,8 +9,22 @@ import {
 
 function* fetchSessionUser() {
   try {
-    const { data: user } = yield call(getSessionUser);
+    const { res, timeout } = yield race({
+      res: call(getSessionUser),
+      timeout: delay(5000)
+    });
 
+    if (timeout) {
+      throw new Error('Request timed out after 5 seconds');
+    }
+
+    if (!res.response.ok) {
```

Comment:
I am not sure if I am testing this poorly, but I added `res.code(500); return res.reply("Bad request");` to the `/user/get-session-user` endpoint, and navigates to `/settings`.

It is stuck on:

<img width="2879" height="1208" alt="image" src="https://github.com/user-attachments/assets/e18a099f-62f3-4eda-94f6-53d9d03ed9c7" />

Reply:
`/settings` is also messed up, this PR doesn't address that.

I believe Tom's working on fixing it, though.